test/unit/io/alignment_file/alignment_file_format_test_template.hpp

```diff
@@ -141,51 +155,46 @@ TYPED_TEST_P(alignment_file_read, input_concept)
 
 TYPED_TEST_P(alignment_file_read, header_sucess)
 {
-    detail::alignment_file_input_format<TypeParam> format;
     typename TestFixture::stream_type istream{this->big_header_input};
 
-    alignment_file_header header{};
-
-    ASSERT_NO_THROW(format.read(istream, input_options, std::ignore, header,  std::ignore, std::ignore, std::ignore,
-                                std::ignore, std::ignore, std::ignore, std::ignore, std::ignore, std::ignore,
-                                std::ignore, std::ignore, std::ignore, std::ignore, std::ignore));
-
-    EXPECT_EQ(header.format_version, "1.6");
-    EXPECT_EQ(header.sorting, "coordinate");
-    EXPECT_EQ(header.subsorting, "coordinate:queryname");
-    EXPECT_EQ(header.grouping, "none");
-
-    EXPECT_EQ(header.program_infos[0].id, "qc");
-    EXPECT_EQ(header.program_infos[0].name, "quality_control");
-    EXPECT_EQ(header.program_infos[0].version, "1.0.0");
-    EXPECT_EQ(header.program_infos[0].description, "trim reads with low qual");
-    EXPECT_EQ(header.program_infos[0].previous, "");
-    EXPECT_EQ(header.program_infos[0].command_line_call, "qc -f file1");
-    EXPECT_EQ(header.program_infos[1].id, "novoalign");
-    EXPECT_EQ(header.program_infos[1].name, "novoalign");
-    EXPECT_EQ(header.program_infos[1].version, "V3.02.07");
-    EXPECT_EQ(header.program_infos[1].description, "");
-    EXPECT_EQ(header.program_infos[1].previous, "qc");
-    EXPECT_EQ(header.program_infos[1].command_line_call, "novoalign -d /path/hs37d5.ndx -f /path/file.fastq.gz");
+    alignment_file_input fin{istream, TypeParam{}};
+
```

Comment:
why not introduce to minimize number of line changes?

```suggestion
    auto header = fin.header();
```